### DISCUSSION TITLE
Feature Requests: timeline navigation and editing

### DISCUSSION THREAD

**Discussion by tingham** [keiwando/toonsquid#75] [Ideas]
CASEYA: When context navigating on a timeline layer or symbol, I would like to be able to quickly scrub the nearest keyframe, either forward or backward, from a" nearest KEY, or FRAME" context menu item. This might also be on the playback button group before and after the next/previous frame buttons. The icons would be the existing arrows, with the keyframe icon overlayed. Realizing that there are "Frames" as well as "Keyframes" it seems like you should be able to start from frame 0; and find the first occurrence of data in the selected layer, then move the timeline cursor.

CASEYB: In the timeline view, a rapid pinch or bloat gesture should scale the timeline to fit or be actual size respectively.

CASEYC: When opening the context menu on a clip in the timeline I would like an option to align the clip: left, center, right - to a specific frame number.

CASEYD: If I have the clips of two timeline layers selected, would like to be able to align left, right, justify the shorter or longer clip to match it's peer.

CASEYE: Given the limited input options available on the iPad; I’d like to have the option to display large chevrons on the left and right of the canvas area, optionally vertically centered, top or bottom aligned. Tapping on the right chevron would advance a single frame; tapping on the left chevron would regress a single frame. In theory this sounds right (as it would work like a carousel) but it might make more sense to have the two buttons always on a preferred-hand side. These buttons would do the same thing that the current next/previous buttons do. They would be available so that you can keep the timeline closed but quickly advance to the next frame to stay focused on the canvas area.

CASEYF: Making a new symbol from the timeline should absolutely adopt the name of the layer it was created from; particularly if that layer has been renamed from the default “Layer #” text. It might also be nice to have an option to "Make Symbol and Edit" that drops you into a focused view on that symbol straight away; separate from the existing menu item.

CASEYG: An option on a timeline layer should be available for “static” or “persistent” so that we don’t have to keep extending the clip duration for those layers, for background and foreground items that don’t change much. Static timeline layers would resize to always match the IN/OUT on the timeline (or just be present for the entire run of the timeline; whichever is easier.)

CASEYH: An alternate view style for the timeline would be welcome; where only the currently selected (active) layer displays as thumbnails. Other layers would display as solid blocks for their clips; with the row height squashed so that we could see more of our view hierarchy at one time. It might also be nice to optional allow for displaying all layers as slabs; without the focused full-size view. Economizing the space in the timeline would be a huge advantage over other animation tools currently available (including those used by major studios.) I tried them all this weekend and they are pretty much universally bad at managing the timeline display, few options and far between.

CASEYI: An optional view style for the timeline would be welcome that places the timeline to the right of the main editing area. This would assist tremendously in modifying larger complex scenes without having to nest things rather enigmatically inside of symbols.

**Comment by keiwando** [comment]
Thank you very much for all of your suggestions!
Just to clarify, when you say "keyframe" do you mean the property keyframes or the first frame of a new drawing?
And the chevron buttons that you mention, would they behave differently than the "forward/backward" buttons on the timeline right now? (https://toonsquid.com/handbook/interface/timeline/#next-previous-frame)

**Reply by tingham** [reply]
@keiwando I've updated my original post. Please let me know if you'd prefer separate issues. I also annotated each item with a prefix in case you have further questions.

Thanks for this awesome app! It's so much nicer than Clip Studio.
